scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1052,7 +1052,36 @@ def cmake(self, filter_stages=[]):
         return self.run_cmake(args,filter_stages)
 
     def build(self):
-        return self.run_build(['--build', self.build_dir])
+        build_result = self.run_build(['--build', self.build_dir])
+        if self.testsuite.bsim_copy_exe and 'BSIM_OUT_PATH' in self.platform.env:
+            self._copy_bsim_executable()
+        return build_result
+
+    def _copy_bsim_executable(self):
+        """
+        Copying application executable to BabbleSim bin directory is necessary
```

Comment:
Maybe better:
"Copying the application executable to BabbleSim's bin directory enables
running multidevice bsim tests after twister has built them".
(?)